tower-http/src/validate_request.rs

```diff
@@ -545,6 +544,25 @@ mod tests {
         assert_eq!(res.status(), StatusCode::OK);
     }
 
+    #[tokio::test]
+    async fn accepted_header_with_quotes() {
```

Comment:
Unless I'm misunderstanding the expected behavior here for the quoted values it would perhaps be nice with an additional test that ensures for example this doesn't pass:

Validate:
``"parisien=\"baguette, jambon, fromage\""``

Incoming req:
`"parisien=\"baguette\""`

Perhaps one variant with vice versa as well

Reply:
I've updated the tests: currently we don't look at the weight, but splitting by `,` would have allowed `text/html` in `parisien=\"baguette, jambon, text/html, fromage\"`. The `mime` iterator handles quoted strings. Does it make more sense now?